[CI] SearchableSnapshotActionIT.testIdenticalSearchableSnapshotActionIsNoop failure

**Build scan**: https://gradle-enterprise.elastic.co/s/yqo5bw6zog6mi/tests/:x-pack:plugin:ilm:qa:multi-node:javaRestTest/org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT/testIdenticalSearchableSnapshotActionIsNoop

**Repro line**:

./gradlew ':x-pack:plugin:ilm:qa:multi-node:javaRestTest' --tests "org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT.testIdenticalSearchableSnapshotActionIsNoop" -Dtests.seed=9C1F76A2EE52C72B -Dtests.security.manager=true -Dtests.locale=sv -Dtests.timezone=America/Managua -Druntime.java=11

**Reproduces locally?**:
No

**Applicable branches**:
master

**Failure history**:
https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=Europe/London&tests.container=org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT&tests.sortField=FAILED&tests.test=testIdenticalSearchableSnapshotActionIsNoop&tests.unstableOnly=true

**Failure excerpt**:
```
java.lang.AssertionError: expected a single document but got: {_shards={total=1, failed=0, successful=1, skipped=0}, count=0} |  
-- | --
  | Expected: <1> |  
  | but: was <0> |  
at __randomizedtesting.SeedInfo.seed([9C1F76A2EE52C72B:6759869CA99195B2]:0) |  
  |   | at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) |  
  |   | at org.junit.Assert.assertThat(Assert.java:956) |  
  |   | at org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT.testIdenticalSearchableSnapshotActionIsNoop(SearchableSnapshotActionIT.java:411)

```

--- a/x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/actions/SearchableSnapshotActionIT.java
+++ b/x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/actions/SearchableSnapshotActionIT.java
@@ -375,11 +375,13 @@ public class SearchableSnapshotActionIT extends ESRestTestCase {
             null
         );
 
-        createIndex(index, Settings.builder()
-            .put(LifecycleSettings.LIFECYCLE_NAME, policy)
-            .build());
+        createIndex(index, Settings.EMPTY);
         ensureGreen(index);
         indexDocument(client(), index, true);
+
+        // enable ILM after we indexed a document as otherwise ILM might sometimes run so fast the indexDocument call will fail with
+        // `index_not_found_exception`
+        updateIndexSettings(index, Settings.builder().put(LifecycleSettings.LIFECYCLE_NAME, policy));
 
         final String searchableSnapMountedIndexName = (storage == MountSearchableSnapshotRequest.Storage.FULL_COPY ?
             SearchableSnapshotAction.FULL_RESTORED_INDEX_PREFIX : SearchableSnapshotAction.PARTIAL_RESTORED_INDEX_PREFIX) + index;
@@ -425,11 +427,13 @@ public class SearchableSnapshotActionIT extends ESRestTestCase {
             null
         );
 
-        createIndex(index, Settings.builder()
-            .put(LifecycleSettings.LIFECYCLE_NAME, policy)
-            .build());
+        createIndex(index, Settings.EMPTY);
         ensureGreen(index);
         indexDocument(client(), index, true);
+
+        // enable ILM after we indexed a document as otherwise ILM might sometimes run so fast the indexDocument call will fail with
+        // `index_not_found_exception`
+        updateIndexSettings(index, Settings.builder().put(LifecycleSettings.LIFECYCLE_NAME, policy));
 
         final String searchableSnapMountedIndexName = SearchableSnapshotAction.PARTIAL_RESTORED_INDEX_PREFIX +
             SearchableSnapshotAction.FULL_RESTORED_INDEX_PREFIX + index;

